src/lazy/any_encoding.rs

```diff
@@ -62,23 +72,99 @@ impl LazyDecoder for AnyEncoding {
     type SExp<'top> = LazyRawAnySExp<'top>;
     type List<'top> = LazyRawAnyList<'top>;
     type Struct<'top> = LazyRawAnyStruct<'top>;
+    type FieldName<'top> = LazyRawAnyFieldName<'top>;
     type AnnotationsIterator<'top> = RawAnyAnnotationsIterator<'top>;
-    type EExpression<'top> = LazyRawAnyEExpression<'top>;
+    type EExp<'top> = LazyRawAnyEExpression<'top>;
+    type VersionMarker<'top> = LazyRawAnyVersionMarker<'top>;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct LazyRawAnyVersionMarker<'top> {
+    encoding: LazyRawAnyVersionMarkerKind<'top>,
 }
+
+#[derive(Debug, Copy, Clone)]
+pub enum LazyRawAnyVersionMarkerKind<'top> {
+    Text_1_0(LazyRawTextVersionMarker_1_0<'top>),
+    Binary_1_0(LazyRawBinaryVersionMarker_1_0<'top>),
+    Text_1_1(LazyRawTextVersionMarker_1_1<'top>),
+    Binary_1_1(LazyRawBinaryVersionMarker_1_1<'top>),
+}
```

Comment:
🗺️ There's now a per-encoding version marker type that can provide a `Span` of its input bytes. Previously we only surfaced the `(major: u8, minor: u8)` pair, but didn't have information about its encoding (particularly its location).